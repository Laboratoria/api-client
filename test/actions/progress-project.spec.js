import {
  addProgressProject,
  deleteProgressProject,
  getProgressProject,
  updateProgressProject,
} from '../../lib/actions/progress-project';

describe('getProgressProject', () => {
  it('should be a function', () => {
    expect(typeof getProgressProject).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(getProgressProject({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});

describe('addProgressProject', () => {
  it('should be a function', () => {
    expect(typeof addProgressProject).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(addProgressProject({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});

describe('updateProgressProject', () => {
  it('should be a function', () => {
    expect(typeof updateProgressProject).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(updateProgressProject({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});

describe('deleteProgressProject', () => {
  it('should be a function', () => {
    expect(typeof deleteProgressProject).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(deleteProgressProject({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});
