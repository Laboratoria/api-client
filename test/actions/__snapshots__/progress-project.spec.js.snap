// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addProgressProject should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "progress-project-add",
    "namespace": "default",
    "suffix": "PROGRESS_PROJECT_ADD",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROGRESS_PROJECT_ADD",
}
`;

exports[`deleteProgressProject should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "progress-project-delete",
    "namespace": "default",
    "suffix": "PROGRESS_PROJECT_DELETE",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "DELETE",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROGRESS_PROJECT_DELETE",
}
`;

exports[`getProgressProject should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "user/someone/progress",
    "namespace": "default",
    "suffix": "PROGRESS_PROJECT",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROGRESS_PROJECT",
}
`;

exports[`updateProgressProject should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "progress-project-update",
    "namespace": "default",
    "suffix": "PROGRESS_PROJECT_UPDATE",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "PUT",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROGRESS_PROJECT_UPDATE",
}
`;
